server/app/scanpy_engine/scanpy_engine.py

```diff
@@ -172,6 +176,50 @@ def _validate_data_calculations(self):
                 f"`cellxgene prepare --layout {self.layout_method} <datafile>` "
                 f"to solve this problem. ")
 
+    def _IEEE754_special_values_workaround_annotations(self):
+        """
+        TODO: temporary workaround
+
+        Because all floating point data is serialized to JSON, and JSON has no means of representing
+        non-finite, floating point special values (NaN, +/-Infinity, etc), we include this temporary
+        work-around.
+
+        This will likely be removed in the future, contingent upon improved marshalling.
+
+        Where non-finite floating point is present in obs, var or X:
+        * issue a warning to the user that these values will be treated as zeros.
+        * set the value to zero within the in-memory data (self.data)
+        """
+        for ax in Axis:
+            curr_axis = getattr(self.data, str(ax))
+            for ann in curr_axis:
+                dtype = curr_axis[ann].dtype
+                if dtype.kind == 'f':
+                    not_finite = np.isfinite(curr_axis[ann]) == False   # noqa: E712
```

Comment:
np.isfinite checks for both infinity and nans? cool!

Reply:
yes indeed!